rm/rm-client/src/test/java/org/ow2/proactive/resourcemanager/common/event/RMInitialStateTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.resourcemanager.common.event;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+
+public class RMInitialStateTest {
+
+    RMInitialState rmInitialState;
+
+    long counter = 0;
+
+    @Before
+    public void init() {
+        rmInitialState = new RMInitialState();
+
+        rmInitialState.nodeSourceAdded(new RMNodeSourceEvent("LocalNodes", counter++));
+        rmInitialState.nodeAdded(new RMNodeEvent("http://localhost:0000", counter++));
+        rmInitialState.nodeAdded(new RMNodeEvent("http://localhost:0001", counter++));
+
+        rmInitialState.nodeSourceAdded(new RMNodeSourceEvent("AzureNodes", counter++));
+        rmInitialState.nodeAdded(new RMNodeEvent("http://localhost:0002", counter++));
+        rmInitialState.nodeAdded(new RMNodeEvent("http://localhost:0003", counter++));
+        rmInitialState.nodeAdded(new RMNodeEvent("http://localhost:0004", counter++));
+
+    }
+
+    @Test
+    public void testLatestCounter() {
+        Assert.assertEquals(6, rmInitialState.getLatestCounter());
+    }
+
+    @Test
+    public void testCloneAndFilter0() {
```

Comment:
Maybe in place of testCloneAndFilter0-1-2, put more specific test name, so that we will have a hint when they fail :)